File: framework/project/base/Assets/Frameworks/Log/Log.cs
using System;
using UnityEngine;
using UnityEngine.EventSystems;

namespace Frameworks
{
    public class Log
    {
        private static string GetTimeStamp()
        {
            var dt = new DateTime(1970,1,1,0,0,0);
            var ts = DateTime.Now - dt;
            var time = Convert.ToInt64(ts.TotalSeconds);
            var ret = dt.AddSeconds(time).ToLocalTime();
            return ret.ToString();
        }

        public static void D(string tag, params string[] arr)
        {
            var info = $"[DEBUG] [{GetTimeStamp()}] [{tag}] {string.Join(" ", arr)}";
            Debug.Log(info);
        }

        public static void W(string tag, params string[] arr)
        {
            var info = $"[WARNING] [{GetTimeStamp()}] [{tag}] {string.Join(" ", arr)}";
            Debug.LogWarning(info);
        }

        public static void E(string tag, params string[] arr)
        {
            var info = $"[ERROR] [{GetTimeStamp()}] [{tag}] {string.Join(" ", arr)}";
            Debug.LogError(info);
        }
    }
}

File: framework/project/base/XScripts/Frameworks/Log.lua
---
--- FileName: Log.lua
--- Created by lu ning
--- DateTime: 2020/07/18 15:25
--- Desc:
---

local Log = {}

local function GetTimeStamp()
    return os.date("%Y/%m/%d %H-%M-%S", os.time())
end

local function FormatLogs(...)
    local arr = { ... }
    if not arr or #arr <= 0 then
        return ""
    end
    local info = ""
    for _,v in ipairs(arr) do
        info = info .. " " .. v
    end
    return info
end

function Log.D(tag, ...)
    local info = "[DEBUG] [" .. GetTimeStamp() .. "] " .. " [" .. tag .. "] " .. FormatLogs(...)
    print(info)
end
function Log.E(tag, ...)
    local info = "[ERROR] [" .. GetTimeStamp() .. "] " .. " [" .. tag .. "] " .. FormatLogs(...)
    print(info)
end
function Log.W(tag, ...)
    local info = "[WARNING] [" .. GetTimeStamp() .. "] " .. " [" .. tag .. "] " .. FormatLogs(...)
    print(info)
end

return Log